csharp-design-patterns/AggregateDesign/AggregateDesign/Domain/Order.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Collections.ObjectModel;
+
+namespace AggregateDesign.Domain;
+
+public enum OrderStatus { PendingPayment, ReadyForShipping, InTransit }
+
+public class Order
+{
+    internal List<OrderItem> _items = new();
```

Comment:
Code external to the aggregate should not be able to access the collection directly. Ideally, it would be a private field but, then you need the data access layer to be able to map data directly into it.

The trade-off is to make it internal since AutoMapper can handle internal fields. I added a sentence to the article explaining this.

Reply:
Thanks for the thorough explanation 😄